src/shared_modules/keystore/include/keyStore.hpp

```diff
@@ -39,7 +68,32 @@ class Keystore final
      * @param key The key to be inserted or updated.
      * @param value The corresponding value to be returned.
      */
-    static void get(const std::string& columnFamily, const std::string& key, std::string& value);
+    static void get(const std::string& columnFamily, const std::string& key, std::string& value)
+    {
+        std::string encryptedValue;
+
+        // Get from DB
+        Utils::RocksDBWrapper keystoreDB = Utils::RocksDBWrapper(DATABASE_PATH, false);
+
+        if (!keystoreDB.columnExists(columnFamily))
+        {
+            keystoreDB.createColumn(columnFamily);
+        }
+
+        if (keystoreDB.get(key, encryptedValue, columnFamily))
+        {
+            if (!std::filesystem::exists(PRIVATE_KEY_FILE))
+            {
+                logWarn(KS_NAME, "No private key was found.");
+                return;
+            }
+            // Decrypt value
+            m_openssl->rsaDecrypt(PRIVATE_KEY_FILE, encryptedValue, value);
+        }
+    }
 };
 
+template<class TOSSLPrimitive>
+std::shared_ptr<TOSSLPrimitive> Keystore<TOSSLPrimitive>::m_openssl {std::make_shared<TOSSLPrimitive>()};
+
```

Comment:
class with static methods with static member, this not make sense.

```suggestion
using Keystore = TKeystore<>;

```